File: src/main/java/no/nav/pensjonbrevdata/BrevdataProvider.java
package no.nav.pensjonbrevdata;

import no.nav.pensjonbrevdata.mappers.brevdata.BrevdataMapper;
import no.nav.pensjonbrevdata.mappers.sakBrev.SakBrevMapper;
import no.nav.pensjonbrevdata.model.Brevdata;
import no.nav.pensjonbrevdata.model.codes.DokumentkategoriCode;
import no.nav.pensjonbrevdata.model.codes.SprakCode;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.ArrayList;
import java.util.List;
import java.util.stream.Collectors;

@Service
public class BrevdataProvider {
    private final BrevdataMapper brevdataMapper;
    private final SakBrevMapper sakBrevMapper;

    @Autowired
    public BrevdataProvider(BrevdataMapper brevdataMapper, SakBrevMapper sakBrevMapper) {
        this.brevdataMapper = brevdataMapper;
        this.sakBrevMapper = sakBrevMapper;
    }

    public List<SprakCode> getSprakForBrevkode(String brevkode) {
        return brevdataMapper.map(brevkode).getSprak();
    }

    public Brevdata getBrevForBrevkode(String brevkode) {
        return brevdataMapper.map(brevkode);
    }

    public List<Brevdata> getBrevdataForSaktype(String saktype){
        return sakBrevMapper.map(saktype).stream()
                .map(brevdataMapper::map)
                .collect(Collectors.toList());
    }

    public List<String> getBrevkoderForSaktype(String saktype) {
        return sakBrevMapper.map(saktype);
    }

    public List<Brevdata> getAllBrev() {
        return brevdataMapper.getAllBrevAsList();
    }

    public List<String> getBrevKeysForBrevkodeIBrevsystem(String brevkodeIBrevsystem) {
        return brevdataMapper.getBrevKeysForBrevkodeIBrevsystem(brevkodeIBrevsystem);
    }

    public List<Brevdata> getEblanketter() {
        return brevdataMapper.getAllBrevAsList().stream()
                .filter(brev -> brev.getDokumentkategori() == DokumentkategoriCode.E_BLANKETT)
                .toList();
    }
}

File: src/main/java/no/nav/pensjonbrevdata/BrevdataProvider.kt
package no.nav.pensjonbrevdata

import no.nav.pensjonbrevdata.mappers.brevdata.BrevdataMapper
import no.nav.pensjonbrevdata.mappers.sakBrev.SakBrevMapper
import no.nav.pensjonbrevdata.model.Brevdata
import no.nav.pensjonbrevdata.model.codes.DokumentkategoriCode
import no.nav.pensjonbrevdata.model.codes.SprakCode
import kotlin.collections.map

class BrevdataProvider(
    private val brevdataMapper: BrevdataMapper,
    private val sakBrevMapper: SakBrevMapper
) {
    fun getSprakForBrevkode(brevkode: String): List<SprakCode>? = brevdataMapper.map(brevkode).sprak

    fun getBrevForBrevkode(brevkode: String): Brevdata = brevdataMapper.map(brevkode)

    fun getBrevdataForSaktype(saktype: String): List<Brevdata> = sakBrevMapper.map(saktype).map { brevdataMapper.map(it) }

    fun getBrevkoderForSaktype(saktype: String): List<String> = sakBrevMapper.map(saktype)

    val allBrev: List<Brevdata>
        get() = brevdataMapper.allBrevAsList

    fun getBrevKeysForBrevkodeIBrevsystem(brevkodeIBrevsystem: String): List<String> = brevdataMapper.getBrevKeysForBrevkodeIBrevsystem(brevkodeIBrevsystem)

    val eblanketter: List<Brevdata>
        get() = brevdataMapper.allBrevAsList.filter { brev -> brev.dokumentkategori == DokumentkategoriCode.E_BLANKETT }
}
